dashboard/utils.py

```diff
@@ -38,6 +44,7 @@ class MMTrack:
     financial_aid_min_price = None
     financial_aid_max_price = None
     financial_aid_date_documents_sent = None
+    final_grades = {}
```

Comment:
using the django cache frameworks adds a roundtrip to the backend system and I am not sure it is worth it if we move to a not in memory cache. (now it is in memory cache)
not sure: your call4